scripts/magic-gen/magicgen.py

```diff
@@ -0,0 +1,77 @@
+import sys
+import re
+import argparse
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='A script to generate Go bindings for the CRIU magic values provided by magic.h')
+    parser.add_argument(
+        'src', help='Input path (magic.h)', type=str)
+    parser.add_argument(
+        'dest', help='Output path (magic.go)', type=str)
+
+    args = parser.parse_args()
+
+    try:
+        with open(args.src, 'r') as src:
+            magics = read_magics(src)
+    except IOError:
+        sys.exit("Failed to open magic.h")
+
+    try:
+        with open(args.dest, 'w') as dest:
+            write_magics(dest, magics)
+    except IOError:
+        sys.exit("Failed to open magic.py")
+
+def read_magics(src):
+    magics = {}
+    pattern = re.compile(r"#define\s+(\S+)\s+0x([0-9A-Fa-f]+)")
```

Comment:
Could you please add a comment to describe the patterns matched by this regular expression? This will be particularly useful for new contributors who are not familiar with the magic-gen scripts.